lib/src/header_parser/sub_parsers/function_type_param_parser.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+
+import '../clang_bindings/clang_bindings.dart';
+import '../data.dart';
+import '../utils.dart';
+
+/// This type holds the list of `ParmDecl` nodes of a function type declaration.
+class FunctionTypeParams {
+  final List<String> paramNames;
+  final Map<String, CXCursor> params;
```

Comment:
Default implementation of Map in Dart is ordered, so you could technically skip the extra array if you like.

Reply:
Is this a guarantee we can rely on? I don't have a preference here.

Reply:
Well separating concept of names and cursors seems clearer to me. Let's leave it as it is.

Reply:
Yep, maps are ordered (there is a separate data store and index, the index stores the entry-number in the data store). A "traditional" hashmap stores a pointer the object directly in the index and has no separate data store. 😄